### PR TITLE
Appliance: Portable AC

### PR DESCRIPTION
**Pull request by tinukedaya**
#### Summary
Content "Add Portable Air Conditioner appliance"

#### Purpose of change

I got sick of running naked in my base during summer and only found the vehicle cooler as a possible solution to that. So I thought I'll try to add a proper AC, now that we have all the appliances.

#### Describe the solution
JSON scavenging/editing. 
As the bases/rooms/buildings are usually much bigger than cars, I made a recipe that uses 2 vehicle coolers to create a bigger (more efficient) and "portable" appliance version. The recipe also contains some hoses to simulate the exhaust heat being directed out of the room/building. However there is no actual exhaust heat generated. It's omitted as being dissipated outside of the building... No idea how to do it differently. 

#### Describe alternatives you've considered

Using the air conditioners that can be found on the roofs for a "simple deconstruction" and using this item as a base for the appliance... But then I thought that those are most likely big industrial air cons that would actually need some construction changes in the walls and such and thus I decided to make the small portable kind via crafting. 

Maybe a small Air Con furniture can be added to the houses, that can be reused as appliance, but that is something I can't do. 

#### Testing

I created new world, debugged myself the skills and mats to craft it and placed it via the construction menu. When turned on, it produces cold air and uses energy. 

#### Additional context

Basically I have no idea what I'm doing here. So I hope I won't mess it up again...

### CODE REVIEW

**Comment by tinukedaya** [comment]
Alright. From  what I can see I had a mistake in one of the commits (stray comma). I fixed that, but if I read the logs correctly, there's something wrong with linting... And I have no clue how to fix that ATM...

**Comment by tinukedaya** [comment]
yay. Seems like I managed to fix it all and all the checks passed!

**Comment by Jarewill** [comment]
I'm looking through the files and I don't see the actual appliance furniture definition.
Maybe I'm missing something, but I wanted to make sure you didn't forget to include it by accident.
Anyways, cheers for adding this!

**Comment by tinukedaya** [comment]
Ah... you might be onto something... Seems like I DID mess up something... And I don't know what... Need to find the lost file...

**Comment by tinukedaya** [comment]
Oh damn... I guess I can't change the comit's name, right? I accidentally linked one really old pull request with it... Although, when checking it I noticed it's quite common. :D

**Comment by kevingranade** [comment]
Why would you rip an AC unit out of a vehicle to cobble together a portable AC unit when self-contained window units are downright common?

**Comment by tinukedaya** [comment]
Are they? I do not see any in game. The only AC units I see in game are those big ones on top of the buildings. And as I wrote before, those feels more like the big units that are used to whole buildings. 

This is supposed to be small portable AC for a single room... They are quite common IRL, so maybe a non craftable version can be added as well and added to some buildings, but that is something I'm not sure how to do. 

So I decided to do this, because I knew I can do it by scavenging the JSON already there.

**Comment by Night-Pryanik** [comment]
Could you please resolve conflicts?

**Comment by tinukedaya** [comment]
Ok, I'll try to do that when I get home. Though from the quick glance I had here on my phone, I have zero idea what needs to be done. Will see... 🤷‍♂️

**Comment by tinukedaya** [comment]
So... I did something with the conflict and it seems like it's gone.

File: data/json/recipes/recipe_appliance.json [hunk 1]
```diff
@@ -106,6 +106,38 @@
...
     "qualities": [ { "id": "SCREW", "level": 1 } ],
     "proficiencies": [ { "proficiency": "prof_elec_soldering" } ],
-    "components": [ [ [ "frame_wood", 1 ] ], [ [ "battery_charger", 1 ] ] ]
+    "components": [ [ [ "frame_wood", 1 ] ], [ [ "battery_charger", 1 ] ], [ [ "medium_battery_cell", 1 ] ] ]
+  },
+  {
+    "result": "portable_air_conditioner",
+    "type": "recipe",
+    "activity_level": "MODERATE_EXERCISE",
+    "reversible": true,
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_appliance.json:115]: With `"reversible": true`, disassembling this AC will take the same 8 hours as it takes to craft it. I suppose disassembling it should require much less time, like 30 minutes or less.
```suggestion
    "reversible": { "time": "30 m" },
```

Comment (tinukedaya) [reply data/json/recipes/recipe_appliance.json:115]: Honestly, as I was just scavenging other appliances json, I had no clear idea of what this does exactly. Your suggestion makes sense and I agree with that.

File: data/json/recipes/recipe_appliance.json [hunk 2]
```diff
@@ -106,6 +106,38 @@
     "using": [ [ "soldering_standard", 5 ] ],
     "qualities": [ { "id": "SCREW", "level": 1 } ],
     "proficiencies": [ { "proficiency": "prof_elec_soldering" } ],
-    "components": [ [ [ "frame_wood", 1 ] ], [ [ "battery_charger", 1 ] ] ]
+    "components": [ [ [ "frame_wood", 1 ] ], [ [ "battery_charger", 1 ] ], [ [ "medium_battery_cell", 1 ] ] ]
```
Comment (Fris0uman) [inline data/json/recipes/recipe_appliance.json:109]: why modify this unrelated recipe?

File: data/json/furniture_and_terrain/appliances.json [hunk 3]
```diff
@@ -474,6 +474,31 @@
       { "item": "steel_plate", "count": [ 2, 4 ] }
     ]
   },
+  {
+    "type": "vehicle_part",
+    "id": "ap_portable_ac",
+    "name": { "str": "portable air conditioner" },
+    "categories": [ "utility" ],
+    "description": "A modified vehicle-mounted area cooler.  A custom made casing with a hose attached which allows the exhaust heat to be directed out of the room.",
```
Comment (Fris0uman) [inline data/json/furniture_and_terrain/appliances.json:482]: It would make much more sense for this to just be a regular furniture that you can find in store and houses, just like irl, instead of being weird makeshift vehicle thing.

File: data/json/items/appliances.json [hunk 4]
```diff
@@ -194,6 +194,21 @@
     "volume": "400 L",
     "weight": "80 kg"
   },
+  {
+    "type": "GENERIC",
+    "id": "portable_air_conditioner",
+    "name": { "str": "portable air conditioner" },
+    "description": "A modified vehicle-mounted area cooler.  A custom made casing with a hose attached which allows the exhaust heat to be directed out of the room.",
```
Comment (Fris0uman) [inline data/json/items/appliances.json:201]: same comment, juts make it a regular AC and make it spawn in the world

File: data/json/recipes/recipe_appliance.json [hunk 5]
```diff
@@ -106,6 +106,38 @@
...
+  },
+  {
+    "result": "portable_air_conditioner",
```
Comment (Fris0uman) [inline data/json/recipes/recipe_appliance.json:112]: Same retrofiting vehicle AC sounds pretty weird, and should at best only be a secondary way to get a cooler working

**Review by Fris0uman — COMMENTED** [review]
- Make it a regular AC
- Make the regular AC spawn in the world as a furniture and/or and item in the appropirate shops for exemple
- Your alternative idea of getting AC from the roof is better than the weird vehicle thing
- Are you sure this works? I'm pretty sure vehicle cooler right now need a muffler on the vehicle to know where to output the heat, how does this works when the vehicle is just a one tile thing with no muffler anywhere?

File: data/json/recipes/recipe_appliance.json (hunk 2 above)
Comment (tinukedaya) [reply data/json/recipes/recipe_appliance.json:109]: This is a conflict resolution... The unrelated recipe was changed in the meantime. But maybe  that was incorrect way how to fix the conflict?

**Comment by tinukedaya** [comment]
> * Make it a regular AC
> 
>     * Make the regular AC spawn in the world as a furniture and/or and item in the appropirate shops for exemple
> 
>     * Your alternative idea of getting AC from the roof is better than the weird vehicle thing
> 
>     * Are you sure this works? I'm pretty sure vehicle cooler right now need a muffler on the vehicle to know where to output the heat, how does this works when the vehicle is just a one tile thing with no muffler anywhere?

As I mentioned above, I have no idea how to make it regular furniture and how to add it to shops and/or the rooftop idea. That is just way above I can/want do so someone else would need to do that. 

I was able to scavenge the appliances json for code that made this work (and yes it does work without the muffler, and I added some description and hose and stuff to abstract the muffler away). 
So basically, this is the way I can contribute with something I know is missing so I did... if it's deemed not suitable, feel free to close this pull request.

File: data/json/recipes/recipe_appliance.json (hunk 2 above)
Comment (Fris0uman) [reply data/json/recipes/recipe_appliance.json:109]: Not sure what you did but this is not right

**Comment by I-am-Erk** [comment]
I think the work you're going for here is great, but you've come at it from an odd direction and have some major issues. At a brief glance you could keep some of this JSON and repurpose it, but it might be easier to close and start again. Your best bet is to look at how fridges and ovens work: they are based on pieces of furniture, not vehicle parts. The vehicle AC is the code you'll want to turn the furniture wall AC into an appliance, but you don't actually want the literal vehicle part in game, just the code from it copied onto the powered furniture.

It's up to you if you prefer to fix this PR or close it and restart but I'd recommend actually restarting and not getting discouraged, you're not actually that far off.

**Comment by tinukedaya** [comment]
Alright. I'll just close this. If anyone wants to pickup and amend it to the above sugestions feel free. I'm not going to do that as I do not have the time to do the research on it.
